plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -257,4 +263,32 @@ protected static List<Column> toHiveColumns(List<NestedField> columns)
                         Optional.empty()))
                 .collect(toImmutableList());
     }
+
+    protected CommitStatus checkCommitStatus(String metadataLocation)
+    {
+        AtomicReference<CommitStatus> status = new AtomicReference<>(CommitStatus.UNKNOWN);
+
+        Tasks.foreach(metadataLocation)
+                .retry(3)
+                .exponentialBackoff(100, 5000, 600000, 4.0)
+                .run(
+                        location -> {
+                            TableMetadata metadata = refresh();
+                            if (metadata != null) {
+                                boolean success = metadata.metadataFileLocation().equals(metadataLocation)
+                                        || metadata.previousFiles().stream().anyMatch(log -> log.file().equals(metadataLocation));
+                                if (success) {
+                                    status.set(CommitStatus.SUCCESS);
+                                }
+                            }
+                        });
+        return status.get();
+    }
+
+    protected void cleanupMetadata(String metadataLocation, CommitStatus commitStatus)
+    {
+        if (commitStatus == CommitStatus.FAILURE) {
```

Comment:
Currently, the cleanup of metadata file happens only when `commitStatus` is `FAILURE`. Should we consider removing the metadata file even when `commitStatus` is `UNKNOWN`?

Reply:
The commit status `UNKNOWN` means that the transaction may or may have not succeeded.

See for example this situation when committing to an existing table:

https://github.com/trinodb/trino/blob/7399e9e418d0b5577809ac4929a7047c4fe07113/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java#L96-L99

If we remove the files we may end up in the situation where the metastore contains information about a table which has no actual content on storage.

Reply:
Thanks, @findinpath for the confirmation. I have asked the same question in the iceberg community via https://github.com/apache/iceberg/pull/6499. Here is the response.

> When the commitStatus is UNKNOWN then we sign the user that the code was not able to decide how could it fix the state, and expects the user to fix the table. In this case we keep everything and expect the user to cleanup the state.